source/validate_cfg.cpp

```diff
@@ -537,39 +543,40 @@ spv_result_t PerformCfgChecks(ValidationState_t& _) {
   return SPV_SUCCESS;
 }
 
-spv_result_t CfgPass(ValidationState_t& _,
-                     const spv_parsed_instruction_t* inst) {
-  SpvOp opcode = static_cast<SpvOp>(inst->opcode);
+spv_result_t CfgPass(ValidationState_t& _, const Instruction* inst) {
+  SpvOp opcode = static_cast<SpvOp>(inst->opcode());
   switch (opcode) {
     case SpvOpLabel:
-      if (auto error = _.current_function().RegisterBlock(inst->result_id))
+      if (auto error = _.current_function().RegisterBlock(inst->id()))
         return error;
+
+      _.current_function().current_block()->set_initiator(inst);
```

Comment:
This creates an odd disagreement in the validator. Due to the order of passes, OpLabel is *not* part of the basic block it starts. Can you add a TODO to fix that and file an issue?

Reply:
Done #1661